### PR TITLE
VIH-8213 updated condition for isavailable flag

### PR DESCRIPTION
**Pull request by omio-ma**
### JIRA link ###
https://tools.hmcts.net/jira/browse/VIH-8213

### Change description ###
- updated is available flag to support quick link users in consultation state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/participant-panel.model.spec.ts [hunk 1]
```diff
@@ -38,6 +38,20 @@ describe('ParticipantPanelModel', () => {
...
+        model = mapper.mapFromParticipantUserResponse(participant);
+        expect(model.isAvailable()).toBeTruthy();
+    });
+
+    it('returns isAvailable: true when participant is quick link observer and in consultation', () => {
+        participant.status = ParticipantStatus.InConsultation;
+        participant.role = Role.QuickLinkParticipant;
+        model = mapper.mapFromParticipantUserResponse(participant);
+        expect(model.isAvailable()).toBeTruthy();
+    });
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/participant-panel.model.spec.ts:53]: Add tests for when is a quicklink user but not in consultation and other roles when in consultation.

Although I would prefer to spyOnProperty isQuickLinkUser so these tests are focused on this specific logic and let the this.isQuickLinkUser tests cover the specific roles.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/participant-panel.model.spec.ts [hunk 2]
```diff
@@ -3,8 +3,9 @@ import { ConferenceTestData } from 'src/app/testing/mocks/data/conference-test-d
 import { ParticipantPanelModel } from './participant-panel-model';
 import { ParticipantPanelModelMapper } from '../../shared/mappers/participant-panel-model-mapper';
 import { HearingRole } from './hearing-role-model';
+import { PanelModel } from './panel-model-base';
 
-describe('ParticipantPanelModel', () => {
+fdescribe('ParticipantPanelModel', () => {
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/participant-panel.model.spec.ts:8]: Remove test focus.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/participant-panel.model.spec.ts [hunk 3]
```diff
@@ -38,20 +39,27 @@ describe('ParticipantPanelModel', () => {
...
+        spyOnProperty(model, 'isQuickLinkUser').and.returnValue(true);
         expect(model.isAvailable()).toBeTruthy();
     });
 
+    it('returns isAvailable: false when participant is quick link user and status is in hearing', () => {
+        participant.status = ParticipantStatus.InHearing;
+        model = mapper.mapFromParticipantUserResponse(participant);
+        spyOnProperty(model, 'isQuickLinkUser').and.returnValue(true);
+        expect(model.isAvailable()).toBeFalsy();
+    });
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/participant-panel.model.spec.ts:61]: One more for isQuickLinkUser is false and inConsultation should return false.

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/participant-panel.model.spec.ts [hunk 4]
```diff
@@ -38,6 +39,27 @@ describe('ParticipantPanelModel', () => {
         expect(model.isAvailable()).toBeTruthy();
     });
 
+    it('returns isAvailable: true when participant is quick link user and status is in consultation', () => {
+        participant.status = ParticipantStatus.InConsultation;
+        model = mapper.mapFromParticipantUserResponse(participant);
+        spyOnProperty(model, 'isQuickLinkUser').and.returnValue(true);
+        expect(model.isAvailable()).toBeTruthy();
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/models/participant-panel.model.spec.ts:46]: Use toBe(true) or toBeTrue()(and toBe(false) or toBeFalse()) as opposed to Truthy and Falsy when testing a boolean

**Review by FarronF — APPROVED** [review]

**Review by krishnapottekkat-hmcts — APPROVED** [review]

**Pull request merged**